### DISCUSSION TITLE
Dealing with multiple groups of lights, inconsistent turn off/on

### DISCUSSION THREAD

**Discussion by gouthamravee** [basnijholt/adaptive-lighting#983] [Q&A]
All the lights in my kitchen are tunable white, but they aren't all the same. 

I have 4 zigbee ledvance recessed lights, 3 zigbee ikea tradfri pendant lights, and 1 LED strip using a shelly RGBW2 as the controller. 
The recessed lights are in their own group in HA, same for the pendant lights, and the LED strip is a light template. 
I have all three groups in one overall group called kitchen lights. 
I have an adaptive lights config specifically for the kitchen lights, and I've added the overall group "kitchen lights" to it. 

The problem is every now and then when I turn off the lights one set of lights doesn't turn off, its usually the pendent lights. 

I've tried a bunch of different configs and nothing seems to work correctly. The biggest issue is if I change the last few settings in the adaptive lights config, one set of lights doesn't work. For instance if I turn off "separate_turn_on_commands" the LED strip light won't respond to light turn on/off events. 

When I added each individual light to the kitchen adaptive light config, the lights turned off one by one and in many cases at least two or three random lights would stay on. 

This also only seems to be a problem when the lights are turned on and off through an automation. Since the lights are smart, I'm using a shelly1pm on the light switch to control the state of the lights. The shelly is setup as an edge switch so it doesn't trip the relay, which stays on all the time. Instead the state of the input on the shelly triggers an automation that changes the state of the overall kitchen group in HA.

I also have the problem that the three sets of lights don't all have the same lower and upper limits for color temp, so I'm not sure how best to group them all for consistent color look through the day. 

Any help with how to configure such a setup to consistently turn all the lights on and off?

**Comment by gravityrebel** [comment]
Just some thoughts:
First, did you read the bit about Ikea bulbs in particular? They are known to have issues. 
Second, and I'm just going based on my own config here so take it with a grain of salt, I'm not using HA groups in Adaptive Lighting. I'm using entities directly. The Adaptive Lighting instructions kind of calls this out specifically for zigbee.
 > For most Zigbee networks, using groups is essential for optimal performance. For example, if you want to use Adaptive Lighting in a hallway with six bulbs, adding each bulb individually to the Adaptive Lighting configuration could overwhelm the network with commands. Instead, create a group in your Zigbee software (not a regular Home Assistant group) and add that single group to the Adaptive Lighting configuration. This sends a single broadcast command to adjust all bulbs, improving response times and keeping the bulbs in sync.

The zigbee group is then exposed to HA as a single entity. For my Hue lights (which are zigbee), I only use the groups. Individual bulbs are all disabled. Even in my entrance where I only have one bulb, I still use a zigbee group. Using the zigbee group will solve the issue of them turning on and off one-by-one. This isn't an issue with AL, but the fact that HA is issuing individual commands to each individual thing in your HA group. The Zigbee group will issue a single broadcast command and all members will respond at the same time. So maybe create three different zigbee groups: recessed, pendant, recessed+pendant. Use those three entities depending on what you want to do instead.

One of the issues with Ikea lights seems to be that they do not respond to commands if they are actively doing something. So if you call "turn off" while they are adapting color they won't see the command. So you can decrease the frequency at which the lights adapt and that will decrease, but not fully eliminate, the potential issue. Or split up your AL config and separate the pendant lights from others. See if isolating the options required for the Ikea bulbs to apply to only the Ikea bulbs helps smooth out your experience. You could also write an that triggers when an "off" command is sent to the pendant group and have it re-check the state every five seconds. If the state is still on, try sending the off command again.

**Comment by gouthamravee** [comment]
Hey @gravityrebel this was the problem for me "One of the issues with Ikea lights seems to be that they do not respond to commands if they are actively doing something." 

I modified the AL settings to not send out too many concurrent messages to the lights. 

You're absolutely right on everything else and I recently re-read the whole AL README and decided I need to re-do my config for these and other similar lights. I plan to do that next month.

**Reply by gravityrebel** [reply]
Nice! Good luck with that. Just be sure to make the changes to everywhere the lights are used, not just in AL. It will dramatically improve your performance and dramatically decrease the amount of chatter on the network.

**Reply by gouthamravee** [reply]
Hey so apparently I was already using zigbee groups, through ZHA's groups function. It also seems like the latest update re-introduced issues for me. I'm re-opening this issue so there's not too much clutter. 

What's happening now is intermittent, seems like there might be a vague pattern. 
When I flip the switch on, usually all the lights come on but sometimes one or two lights will lag behind. When I turn off the switch, again most times they'll all turn off, but sometimes one or two lights will lag behind. 

Now maybe 1 out of every 10/12 turn on/off switches one or two lights won't respond at all. Either they won't turn off, or they won't turn on. 

I have 1 zigbee group that's 7 lights. 3x Ikea Tradfri and 4x Ledvance recessed. 
There's one more light entity in the kitchen, its a custom HA entity that controls a Shelly RGBW2. 

I've tried this three ways; 

1. Two separate AL configs, one for the zigbee light group and one for the RGBW2 light entity. 

2. One AL config for the kitchen, with the Zigbee group and the RGBW2 entity as separate entities. 

3. A HA group with the zigbee group and RGBW2 entity, then the group added to a single AL config. 

All three have the same problem. 

The issue is intermittent, everything will work fine for days or weeks and then all of a sudden things will start acting weird. 

As far as I can tell the network strength between all of these is great, all green in the ZHA visualization. 

I also don't have similar problems with other zigbee groups.
